### PR TITLE
Rspec Pull Request

### PR DESCRIPTION
**Pull request by Usagib**
Pull request open for TDD project

### CODE REVIEW

**Comment by Usagib** [comment]
Hello Team!
We made the tests on my enumerable project repo
Sending to code review
Happy Coding! and Happy new Year!

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Good job :+1: 
You have done well in this project. All your methods pass the tests. But we need to fix a few minor things before we can merge this PR.
Check my comments.

### Martin Afani

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,224 @@
+#spec/enumerable_spec.rb
+require './lib/enumerable'
+
+RSpec.describe Enumerable do
+  test = [2,3,4,5,6,7]
+  testeven = [2,4,6,8,10]
+  testodd = [1,3,5,7,9]
+  hash = {"one" => 1, "two" => 2, "three" => 3}
+  testclass = Class.new
```
Comment (whiz25) [inline spec/enumerable_spec.rb]: - [x] it's recommended to declare the variables like;
`let(:test){[2,3,4,5,6,7]},
  let(:testeven){[2,4,6,8,10]},
  let(:testodd){[1,3,5,7,9]},
  let(:hash){{"one" => 1, "two" => 2, "three" => 3}},
  let(:testclass){Class.new}`

File: README.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
 # Ruby Enumerable methods
 A comprehensive rebuild on basic Enumerable methods in Ruby for a better understanding in block passing, enumerator and classes and regular expressions as arguments.
 
+## Update: Added Rspec testing for all methods, check ./spec/enumerable_spec.rb for details.
+
+### Testing project specification can be found at: https://www.theodinproject.com/courses/ruby-programming/lessons/testing-your-ruby-code
+
 ### This scripts rebuilds ruby Enumerable methods:
```
Comment (whiz25) [inline README.md]: - [x] The project should have a descriptive README.md ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

**Comment by Usagib** [comment]
Hello team!, done with the corrections
happy coding

**Review by stratospherique — APPROVED** [review]
### Great Job applying the suggested changes :clap: :clap: The project is approved :heavy_check_mark: You may merge :+1: :+1: :+1: 

Keep up the good work :muscle: 
**_Ahmed Mahfoudh_**

**Pull request merged**